src/VisualStudio/VisualStudioDiagnosticsToolWindow/OptionPages/PerformanceLoggersPage.cs

```diff
@@ -47,19 +48,19 @@ protected override void OnApply(PageApplyEventArgs e)
             SetRoslynLogger(loggerTypes, () => new TraceLogger(options));
 
             // second set RemoteHost options
-            var client = _remoteHostClientService.TryGetRemoteHostClientAsync(CancellationToken.None).Result;
+            var client = ThreadHelper.JoinableTaskFactory.Run(() => _remoteHostClientService.TryGetRemoteHostClientAsync(CancellationToken.None));
```

Comment:
Why the change?

Reply:
Because calling `Task.Result` (on an incomplete task) will hang if the task requires the UI thread to complete (which by default it will, if started on the UI thread). JTF.Run won't deadlock under the same circumstances.

Reply:
I guess I should have clarified my question: why the change _now_? We have literally hundreds of these in Roslyn, why just this one? :smile:

Reply:
My guess is it's one of the very few that appear in a project that runs the vs-threading analyzers. But @sharwell would have to say for sure.

Reply:
> My guess is it's one of the very few that appear in a project that runs the vs-threading analyzers.

Guess is correct 👍